### PR TITLE
add minor changes for improving states robustness

### PR DESCRIPTION
**Pull request by MalloZup**
This pr is fixing a failure in SLE15 for state when deploying cluster.

Increase timeout for ha-proxy so state doesn't fail

### CODE REVIEW

**Comment by MalloZup** [comment]
yop thx! i will document

File: salt/etcd/init.sls [hunk 1]
```diff
@@ -66,10 +66,6 @@ etcd:
     {%- if should_register_etcd_member %}
       - add-etcd-to-cluster
     {%- endif %}
-  # wait until etcd is actually up and running
-  caasp_etcd.healthy:
```
Comment (ereslibre) [inline salt/etcd/init.sls]: I don't think this should go away. The previous check is just confirming that etcd is running in a stable way in this machine, but this state is making sure the etcd cluster is healthy as a whole, taking all etcd members into account.

Comment (MalloZup) [reply salt/etcd/init.sls]: yop

File: salt/etcd/init.sls [hunk 2]
```diff
@@ -66,10 +66,15 @@ etcd:
...
-  # wait until etcd is actually up and running
+# wait until etcd is actually up and running
   caasp_etcd.healthy:
     - watch:
       - caasp_service: etcd
+    - retry:
+        attempts: 10
+        until: True
+        interval: 60
+        splay: 10
```
Comment (ereslibre) [inline salt/etcd/init.sls]: We can probably just use `attempts` and `interval`. I think this interval is way too high though, what about `15` interval?

Comment (MalloZup) [reply salt/etcd/init.sls]: okl i can try

Comment (MalloZup) [reply salt/etcd/init.sls]: done

File: salt/_modules/caasp_hosts.py [hunk 3]
```diff
@@ -209,8 +209,15 @@ def managed(name=HOSTS_FILE,
     /etc/hosts:
       caasp_hosts.managed
     '''
-    this_roles = __salt__['grains.get']('roles', [])
-    infra_domain = __salt__['caasp_pillar.get'](PILLAR_INTERNAL_INFRA, 'infra.caasp.local')
+    try:
+      this_roles = __salt__['grains.get']('roles', [])
```
Comment (inercia) [inline salt/_modules/caasp_hosts.py]: Indentation seems to be wrong, right? Two spaces here, three later on...

Comment (MalloZup) [reply salt/_modules/caasp_hosts.py]: it is just a wip stuff for debugging, i agree with your pov :grin:

**Comment by MalloZup** [comment]
> I'm not really fond of catching all these exceptions just for printing a `hey, this has failed!` error. I personally prefer to see the backtrace, as everything is collapsing anyway...

yop thx @inercia  for review. This pr contain all this things for debugging an issue we have on sle15 i will remove all this code later

File: salt/etcd/init.sls [hunk 4]
```diff
@@ -66,10 +66,15 @@ etcd:
     {%- if should_register_etcd_member %}
       - add-etcd-to-cluster
     {%- endif %}
-  # wait until etcd is actually up and running
+# wait until etcd is actually up and running
```
Comment (inercia) [inline salt/etcd/init.sls]: ??

File: salt/kube-apiserver/init.sls [hunk 5]
```diff
@@ -40,8 +40,12 @@ kube-apiserver:
...
     - source:     salt://kube-apiserver/apiserver.jinja
     - template:   jinja
-  service.running:
-    - enable:     True
+  caasp_service.running_stable:
+    - name: kube-apiserver
+    - successful_retries_in_a_row: 10
+    - max_retries: 30
+    - delay_between_retries: 2
+    - enable: True
```
Comment (inercia) [inline salt/kube-apiserver/init.sls]: I think indentation is wrong...

File: salt/etcd/init.sls (hunk 4 above)
Comment (MalloZup) [reply salt/etcd/init.sls]: done

File: salt/kube-apiserver/init.sls (hunk 5 above)
Comment (MalloZup) [reply salt/kube-apiserver/init.sls]: let me check

Comment (MalloZup) [reply salt/kube-apiserver/init.sls]: nope is your github maybe?

**Comment by MalloZup** [comment]
@ereslibre i plan to merge this after CI pass

**Pull request merged**